bokeh/_testing/util/filesystem.py

```diff
@@ -143,20 +154,12 @@ async def with_temporary_file_async(func, dir=None, suffix=''):
         f.close()
         os.remove(f.name)
 
-class WorkingDir:
-    '''
-
-    '''
-    def __init__(self, pwd):
-        self._new = pwd
-        self._old = os.getcwd()
-
-    def __exit__(self, type, value, traceback):
-        os.chdir(self._old)
-
-    def __enter__(self):
-        os.chdir(self._new)
-        return self._new
+@contextmanager
+def WorkingDir(new: PathLike) -> Iterator[PathLike]:
+    old = os.getcwd()
+    os.chdir(new)
+    yield new
+    os.chdir(old)
```

Comment:
@mattpap I thought we recently agreed on a different style here https://github.com/bokeh/bokeh/pull/11219#discussion_r623270110

```
def _get_save_args(
    state: State, 
    filename: Optional[str], 
    resources: Optional[Resources],
    title: Optional[str]
) -> Tuple[str, Resources, str]:
```

Reply:
I applied this to some extreme cases, but otherwise it's a lot of rewriting at this point. Let's do that after the initial typing pass is done (which I think can be achieved before 2.4).